esx_service/utils/auth_data.py

```diff
@@ -514,15 +531,30 @@ def connect(self):
         if not self.conn:
             raise DbConnectionError(self.db_path)
         
+        # Return rows as Row instances instead of tuples
+        self.conn.row_factory = sqlite3.Row
+
         if not need_create_table:
             if os.path.isfile(self.db_path) and self.need_upgrade_db():
                 # Currently, when need upgrade, raise exception
                 # TODO: add code to handle DB schema upgrade 
                 # when schema changes
                 raise DbConnectionError(self.db_path)        
+            
+            error_msg, tenant = self.get_tenant(auth.DEFAULT_TENANT)
+            if error_msg:
+                raise DbConnectionError(self.db_path)
+            if not self.is_valid_default_tenant_uuid(tenant.id):
+                error_msg = """
+                            Your ESX installation seems to be using config db created by previous version of
+                            Vsphere Docker Volume Service, and requires upgrade.
+                            Please download script from {0} and run the script on each ESX where you are using
```

Comment:
"Please download script from {0} and run the script on each ESX where you are using the product." =>
"Please download script from {0} and run the script on each ESX where vSphere Docker Volume Service is installed"